extension/encoding/awslogsencodingextension/internal/unmarshaler/elb-access-log/elb.go

```diff
@@ -434,19 +433,36 @@ func scanField(logLine string) (string, string, error) {
 		return value, remaining, nil
 	}
 
-	// if there is a quote, we need to get the rest of the value
-	logLine = logLine[1:] // remove first quote
-	value, remaining, found := strings.Cut(logLine, `"`)
-	if !found {
-		return "", "", fmt.Errorf("value %q has no end quote", logLine)
+	// preserve values without quotes, terminate at space after the closing quote or at line end
+	var quoteStack []rune
+	var unquotedValue []rune
+	for i := 0; i < len(logLine); i++ {
```

Comment:
Can NLB logs contain multi-byte characters? I don't think so... IIANM, internationalized domain names should only show in the logs as Punycode.

Reply:
@axw this was a really good point. I updated `scanField` helper to iterate over runes https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/44109/commits/bfddad508fec9289b85e6003637f859d975fa89b

Also added a test which failed with current implementation. From what I found URL paths and query sections can contain multi byte characters so they can end up in logs.

Reply:
Thanks for looking into it! Change looks good.